crates/ir/src/container.rs

```diff
@@ -0,0 +1,107 @@
+use std::mem;
+
+use crate::{
+    op_code::{IrInstruction, IrOpCode},
+    register::{chill::RegisterChill, RegisterId},
+};
+
+pub struct IrContainer {
+    _data: Vec<u8>,
+}
+
+impl IrContainer {
+    /// # Safety
+    /// This function can receive any data and it is up to the caller to ensure that the data is valid IR in valid
+    /// version.
+    pub unsafe fn from_vec(data: Vec<u8>) -> Self {
+        IrContainer { _data: data }
+    }
+
+    pub fn iter(&self) -> IrContainerIterator {
+        IrContainerIterator {
+            container: self,
+            index: 0,
+        }
+    }
+}
+
+impl<'container> IntoIterator for &'container IrContainer {
+    type Item = IrInstruction<'container>;
+    type IntoIter = IrContainerIterator<'container>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.iter()
+    }
+}
```

Comment:
I think this `IntoInterator` implementation should be removed, because using this iterator is **unsafe** as it calls into unsafe code without checks. This allows for calling into `unsafe` code without any `unsafe` blocks!
```rs
let container: IrContainer = ...;
for element in container {
    // This Is A Bad.
}
```

Instead, make `IrContainer::iter` an `unsafe fn` to imply that using the iterator may result in Undefined Behavior.

Reply:
Why you think using iterator is unsafe?

Reply:
Because the iterator immediately calls an `unsafe` function. (This is assuming we remove the `unsafe` from `new`, which I described my motivation for in a previous comment.)